app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -134,15 +184,85 @@ public void onLocationChanged(Location location) {
         if (location == null) {
             return;
         }
+        double longitudePFL = location.getLongitude() - currentLocation.getLongitude();
+        double latitudePFL = location.getLatitude() - currentLocation.getLatitude();
         currentLocation = location;
-        displayLocation();
+        LatLng speedVector = new LatLng(latitudePFL, longitudePFL);
+        LatLng actualLocation = new LatLng(
+                currentLocation.getLatitude(),
+                currentLocation.getLongitude());
+    }
+
+    private double positiveRemainder(double divisor, double dividend) {
+        double quotient = Math.floor(dividend / divisor);
+        return dividend - divisor * quotient;
+    }
+
+    private HashMap<Integer, ParsePolygon> getActualGridSquare() {
+        HashMap<Integer, ArrayList<ParseGeoPoint>> grid = new HashMap<>();
+        //squareSize is a "debug number and it will change depends of the speedVector"
+        double squareSize = 0.005;
+        int numSquareSizeGrid = 3;
+        double cornerLatitudeCC = currentLocation.getLatitude() - positiveRemainder(squareSize, currentLocation.getLatitude());
```

Comment:
Not sure what CC is, a comment might be helpful here to describe what this is calculating.

Reply:
And why a positive remainder is needed.

Reply:
The CC have a typo, I want to CS that means **C**enter **S**quare of the grid.

And the positive remainder is needed bc I need obtain (if we think with the numeric line) the left close number that his remainder with the square size is 0.